### PR TITLE
VYZN: Highlight selection inside a model when its occulded 

### LINKED ISSUE #454: VYZN: Viewer: Highlighting elements inside a model need to be visible
If I select an element that is inside a model, e.g. a certain room or inside wall, the element highlight is not visible because it is blocked by other elements. Highlighting elements should "shine through" somehow, similar to how Blender does it:

image

### PR DESCRIPTION
**Pull request by aozien**
**This PR closes #454** 
Update: it should also close **#382** and **#550** as well

It's achieved by using a PostProcessing effect to the scene after the initial render pass, I had to override [the render function ](https://github.com/IFCjs/web-ifc-viewer/blob/f8611ef91ba2081959e6531eb98f8c8cec72e14f/viewer/src/components/context/renderer/renderer.ts#L44) created by the IFC-Viewer to be able to create custom render pipeline to avoid duplicate rendering on the scene, as specified [here](https://threejs.org/docs/#manual/en/introduction/How-to-use-post-processing) 

> When using a composer, it's necessary to change the application's animation loop. Instead of calling the render method of [WebGLRenderer](https://threejs.org/docs/index.html#api/en/renderers/WebGLRenderer), we now use the respective counterpart of [EffectComposer](https://threejs.org/docs/index.html#examples/en/postprocessing/EffectComposer).

## how it works
initializing **IFCHighlighter** overrides the (viewer render function) to use the composer render instead, then the highlighter can set what's highlighted based on the current selected Meshes from **IFCSelection**

image

### CODE REVIEW

File: src/Infrastructure/IfcHighLighter.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (pablo-mayrgundter) [inline src/Infrastructure/IfcHighLighter.js:1]: No empty line

File: src/Infrastructure/IfcHighLighter.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    // override the viewer rendering pipeline
+    const renderer = context.getRenderer()
+    const scene = context.getScene()
+    const camera = context.getCamera()
+    const {composer, outlineEffect} = createComposer(renderer, scene, camera)
+    this._outlineEffect = outlineEffect
+    context.renderer.update = newUpdateFunction(context, composer)
+  }
+
+  /**
```
Comment (pablo-mayrgundter) [inline src/Infrastructure/IfcHighLighter.js:30]: 2 empty lines between functions (and throughout)

**Review by pablo-mayrgundter — APPROVED** [review]
Neat!

Needs 2 unit tests for selection/deselection expects to the jest mocks.  And a couple formatting nits.

Thanks!

**Comment by aozien** [comment]
Hi @pablo-mayrgundter, added tests for highlighting, removed empty lines, PTAL

**Pull request merged**